service.watchdog/core/utils.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 Thomas Amland
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+
+import os
+import re
+import sys
+import xbmc
+import xbmcgui
+import json
+from urllib.parse import unquote
+from threading import Condition
+
+
+def log(msg, level=xbmc.LOGDEBUG):
+    from . import settings
+    xbmc.log(("[" + settings.ADDON_ID + "] " + msg).encode('utf-8', 'replace').decode(), level)
+
+
+def encode_path(path):
+    """os.path does not handle unicode properly, i.e. when locale is C, file
+    system encoding is assumed to be ascii. """
+    if sys.platform.startswith('win'):
+        return path
+    return path.encode('utf-8')
+
+
+def decode_path(path):
```

Comment:
Still needed in python3?

Reply:
It's shouldn't be, I can try removing it.